modules/price/src/Price.php

```diff
@@ -247,6 +247,22 @@ public function lessThanOrEqual(Price $price) {
   }
 
   /**
+   * Rounds the price to the specified precision.
+   *
+   * @param string $precision
+   *   The precision.
+   * @param int $mode
+   *   The rounding mode.
+   *
+   * @return static
+   *   The resulting price.
+   */
+  public function round($precision = '2', $mode = PHP_ROUND_HALF_UP) {
```

Comment:
@mglaman 
Yes, the default precision and rounding mode may cause mayhem on some currencies. Think of currencies which don't have decimals (no coins) or which want to have precision more than 2. For example, the [price for gold per ounce](http://www.apmex.com/spotprices/gold-price) is always displayed with  precision 3 even if the decimals are zeroes. In drupal 7 the parameters for each currency is hardcoded and can be fetched by a currency code but that is not a good solution, as I think, because:

1. It forces us to depend on some external resource and to maintain the currencies data up to date.
2. It forces a user to use the precision defined by this resource.

As I said earlier the best way is to make order item price widget configurable with the same default values what you've added to the current round() version. Then invokers can fetch the widget settings and build the price to be used.